lib/firebase.rb

```diff
@@ -53,11 +53,18 @@ def update(path, data, query={})
     private
 
     def process(verb, path, data=nil, query={})
+      if non_relative_path_against_child_base_url?(path)
+        raise(ArgumentError.new("Path invalid: #{path}. A relative path must be used against a child base_uri"))
+      end
       Firebase::Response.new @request.request(verb, "#{path}.json", {
         :body             => (data && data.to_json),
         :query            => (@auth ? { :auth => @auth }.merge(query) : query),
         :follow_redirect  => true
       })
     end
+
+    def non_relative_path_against_child_base_url?(path)
+      path.lstrip[0] == '/' && URI(@request.base_url).path != "/"
```

Comment:
BTW a '/' is always added to the base_uri on line 15 so you wouldn't get an empty string at that point. But I see your point that it's more consistent to always throw an error when an absolute path is used, regardless of whether the base_uri is a root or child reference.